pkg/test/framework/components/echo/kube/deployment.go

```diff
@@ -58,64 +58,66 @@ spec:
   selector:
     app: {{ .Service }}
 ---
+{{$deployConfig := .Workloads }}
```

Comment:
How do you envision this working, exactly? Each deployment will have the same namespace and name, so won't k8s simply overwrite the deployment each time?

Reply:
good question. in my case, i used the `Service-Version` to concatenate the name. My use case just use different versions on different workloads. So the deployment won't conlide.

Maybe we can add another field of `WorkloadConfig` like `Name` and put additional label as `deployment: {Name}` for different selectors? Then we avoid abusing version field.

Reply:
How is that different from how we deploy multiple versions of a service in the tests today? Maybe you should just do that instead?

Reply:
how is multi versions of a service used? could you give a link of code of such usage?

my understanding is that will create different services, one per version. Am I able to use one `echo.Instance.Call` and that Call can end up into different destination pods, one having sidecar one does not?

Reply:
Hmm ... I thought we had examples, but I don't see any of our tests using versions. Traffic shifting, for example, uses separate services.

I think this change is generally ok since if you create multiple echo instances, it would also create several k8s services, which isn't what you want. It would be better to create a single instance with multiple deployments as you've defined here.

Reply:
yup. sounds good.